### PR TITLE
Add plotting options to context menu for WorkspaceGroups

### LINKED ISSUE #27820: Add plotting options to context menu for group workspaces in Workbench
This would provide the same behaviour as MantidPlot. Particularly useful for the advanced plots menu because it is primarily used for plotting multiple workspaces.

### PR DESCRIPTION
**Pull request by PhilColebrooke**
**Description of work.**
This PR makes it so that you can right-click a WorkspaceGroup and plot the workspaces it contains, like in MantidPlot.

**To test:**
1. Make a WorkspaceGroup and right-click it. There should now be plotting options.
2. Plotting this way should produce a plot with lines from each workspace in the group. If one of the workspaces can't be plotted, such as a TableWorkspace, a warning will appear in the log but the rest of the workspaces will still be plotted.
3. A WorkspaceGroup containing only non-plottable workspaces should not have the plotting options when right-clicked.

Fixes #27820 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/python/mantidqt/dialogs/spectraselectordialog.py [hunk 1]
```diff
@@ -49,16 +50,26 @@ def __init__(self, workspaces):
...
-                raise ValueError("Expected MatrixWorkspace, found {}.".format(ws.__class__.__name__))
+            try:
+                if not isinstance(ws, MatrixWorkspace) :
+                    raise ValueError("{}: Expected MatrixWorkspace, found {}".format(ws.name(), ws.__class__.__name__))
+            except ValueError as e:
+                # Log the error but carry on so valid workspaces can be plotted.
+                logger.warning(str(e))
+                continue
+            else:
+                matrix_workspaces.append(ws)
```
Comment (ConorMFinn) [inline qt/python/mantidqt/dialogs/spectraselectordialog.py]: This looks like it could just be `if isinstance: append; else: log warning`.

**Review by cailafinn — CHANGES_REQUESTED** [review]
Show detector table on a workspace group containing only empty table workspaces causes an unhandled exception.

Other than that, just the one code comment.

**Review by cailafinn — DISMISSED** [review]
Code looks good and functions as described. Good work. :+1:

**Review by NickDraper — CHANGES_REQUESTED** [review]
```
No release notes because who cares
```

I have to say I disagree with this pretty unprofessional comment.
This is a user facing GUI change, that if you do not tell people about it will never be used.

**Comment by NickDraper** [comment]
CLang format cancelled as the last build passed, and the final commit was release notes only and not complex.

Merged

**Pull request merged**